Composer/packages/lib/indexers/src/validations/expressionValidation/index.ts

```diff
@@ -49,8 +49,10 @@ export const validateExpressions: ValidateFunc = (
     let errorMessage = '';
     let warningMessage = '';
     try {
-      newCache[value] = cache?.[value] ? cache[value] : checkExpression(value, required, types);
-      errorMessage = checkReturnType(newCache[value], types);
+      // The cacheKey adds the type of the value at the end, this allows a new indexing system to prevent the overwriting in cache.
+      const cacheKey = types.length > 1 ? value : `${value}.${types[0]}`;
```

Comment:
Can we store a type along with the cached result instead and check for types match before returning the value?

My only concern is cache pollution with values which probably won't be used anymore.

Reply:
Hi @OEvgeny, we updated the solution to check for errors before saving the types in the cache.
Please let us know if this was not what you had in mind.
Thanks!

Reply:
Looks good. Please update the PR description with the details

Reply:
Done. Thanks!